### PR TITLE
React Wrapper Step 2: Break RotateContainer and cleardiv out of EJS

### PR DESCRIPTION
**Pull request by islemaster**
Start removing content from page.html.ejs.
- Makes `#rotateContainer` into a component and renders it using React.
- Renders `.clear` div using React.
- Extracts top-level render parameters into variables, [per Brent's suggestion](https://github.com/code-dot-org/code-dot-org/pull/7000#discussion_r54275490) in #7000.
- Updates NetSim to use AppView.jsx for its top-level rendering and removes config.html entirely.  NetSim will need it's own top-level jsx at some point, but hasn't diverged yet.

TODO before merge:
- [x] We show/hide the rotate container in our own code sometimes - move that logic to React.
- [x] Make sure any reference to config.html in Ruby code has been removed.

_Step 2 from [this design document](https://docs.google.com/document/d/1itHwWa0ilE4FAMjDuqxkWXSqp_d0z7lhTeMFJybFH0w/edit?usp=sharing)._

### CODE REVIEW

**Comment by islemaster** [comment]
CC @bcjordan @Bjvanminnen @joshlory @davidsbailey

**Comment by islemaster** [comment]
Once again, [diff ignoring whitespace](https://github.com/code-dot-org/code-dot-org/pull/7026/files?w=0) is recommended.

**Comment by islemaster** [comment]
Okay, pretty confident `config.html` has never been set from dashboard code and there's nothing to clean up.

> No occurrences of `^\s*html\s*$` found in dashboard.

(If dashboard was setting `config.html` I'd expect to find an entry somewhere like the [serialized_attrs in blockly.rb](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/blockly.rb#L26))

File: apps/src/templates/RotateContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+var RotateContainer = React.createClass({
+  propTypes: {
+    assetUrl: React.PropTypes.func.isRequired
+  },
+
+  render: function () {
+    return (
+      <div id="rotateContainer" style={this.getStyle()}>
+        <div id="rotateText">
+          <p>{ msg.rotateText() }<br />{msg.orientationLock()}</p>
```
Comment (Bjvanminnen) [inline apps/src/templates/RotateContainer.jsx]: Any reason for the whitespace surrounding msg.rotateText() but not msg.orientationLock?

**Comment by Bjvanminnen** [comment]
lgtm

File: apps/src/netsim/netsim.js [hunk 2]
```diff
@@ -224,35 +224,45 @@ NetSim.prototype.init = function(config) {
...
-    data: {
-      visualization: '',
-      localeDirection: this.studioApp_.localeDirection(),
-      controls: require('./controls.html.ejs')({assetUrl: this.studioApp_.assetUrl})
-    },
-    hideRunButton: true
-  });
+  var renderCodeApp = function () {
+    return page({
+      assetUrl: this.studioApp_.assetUrl,
```
Comment (Hamms) [inline apps/src/netsim/netsim.js]: does page still need assetUrl?

**Comment by Hamms** [comment]
LGTM, pending that remaining TODO

File: apps/src/templates/RotateContainer.jsx (hunk 1 above)
Comment (islemaster) [inline apps/src/templates/RotateContainer.jsx]: :dizzy_face: :palm_tree: : No.

File: apps/src/netsim/netsim.js (hunk 2 above)
Comment (islemaster) [inline apps/src/netsim/netsim.js]: Nice catch.  NetSim's page.html.ejs does not (though the generic one still does).

**Comment by islemaster** [comment]
@Hamms @Bjvanminnen PTAL:
- Responded to CR feedback
- Moved `RotateContainer` show/hide to React rendering.

The latter step has actually duplicated knowledge of whether to require landscape mode across the different apps.  I'm not sure where to deduplicate it again.
1. `StudioApp` could get a static `requireLandscape(config)` helper
2. `AppView` could require the `share` and `config` props separately, and own the `requireLandscape` logic internally
3. Dashboard could compute `requireLandscape` itself and put it on the config object, which would get passed bare to `AppView`
4. Most dramatic: We could move top-level rendering responsibility back into `StudioApp` but have the specific app pass a content component (where the `ProtectedStatefulDiv` goes now).

Thoughts? I'm learning toward option 2 right now, but not sure any are worth doing yet - it's possible the next step (splitting page.html.ejs) will bring more clarity to this discussion.

**Comment by Bjvanminnen** [comment]
2 sounds reasonable to me, but punting on all for the time being also seems fine

**Comment by islemaster** [comment]
Yeah, punting on all for now.

**Pull request merged**